content/6_Advanced/Parallel_Binary_Search.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: parallel-binary-search
+title: 'Parallel Binary Search'
+author: Muaath Alqarni
+description: 'Answer binary search queries offline'
+prerequisites:
+  - binary-search
+  - 'Divide & Conquer'
+frequency: 0
+---
+
+
+## Resources
+
+<Resources>
+	<Resource source="CF" title="Parallel Binary Seach [tutorial]" url="45578" />
+	<Resource source="SPBook" title="Parallel Binary Seach" url="https://kostka.dev/sp/spbook.pdf#page=22" />
+</Resources>
+
+## Tutorial
+This technique is usually used when we can solve a single query by doing linear or binary search on monotonic function.
+
+The trick is doing it offline, sort queries by their median (initially $l$, $r$ will be the binary search values), do linear search,
+whenever you find a query its median in same as your position in linear search, check the monotonic function, and do same as binary search (edit either $l$ or $r$).
+
+You will need to do the linear search $O(\log_2{N})$ times, so $l = r$.
+
+Usually the complexity would be: $O((N+Q) \cdot \log_2{(N+Q)}^2)$
```

Comment:
This is an incomplete thought. Usually it's that time complexity but you should also state the time complexity of the parallel binary search following a ", but"